common/model/transfer.c

```diff
@@ -0,0 +1,673 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include "common/model/transfer.h"
+#include <string.h>
+#include "common/helpers/sign.h"
+#include "common/trinary/trit_tryte.h"
+#include "common/trinary/tryte_long.h"
+
+#define TRITS_PER_MESSAGE 6561
+#define TRITS_PER_ESSENCE 486
+#define TRITS_PER_BUNDLE_HASH 243
+#define TRYTES_PER_MESSAGE 2187

```

Comment:
It appears you're using these as offset multiplicators; however, you're using flextrits! That won't work.